### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by aalavandhan**
* Updated aAMPL math to use `int265` rather than `uint256`
* Added aAMPL specific unit tests for borrow, repay, deposit, withdraw and rebase

To run the aAMPL tests
```
yarn run test-ampl
```

### CODE REVIEW

File: contracts/protocol/tokenization/ampl/AAmplToken.sol [hunk 1]
```diff
@@ -622,30 +632,43 @@ contract AAmplToken is VersionedInitializable, IncentivizedERC20, IAToken {
...
+  function _scaledTotalSupply(ExtData memory e, int256 totalGonsDeposited) private pure returns (uint256) {
+    // require(totalGonsDeposited>=e.totalGonsBorrowed);
+    return _gonsToAMPL(e, uint256(totalGonsDeposited.sub(e.totalGonsBorrowed)))
       .add(e.totalPrincipalBorrowed);
   }
 
+  /**
+   * @dev Helper method to convert gons to AMPL
+   **/
+  function _gonsToAMPL(ExtData memory e, uint256 gonValue) private pure returns (uint256) {
```
Comment (ahnaguib) [inline contracts/protocol/tokenization/ampl/AAmplToken.sol]: @nithinkrishna Why this restructure?

Comment (aalavandhan) [reply contracts/protocol/tokenization/ampl/AAmplToken.sol]: For the sake of readability

File: test/ampl-atoken.spec.ts [hunk 2]
```diff
@@ -910,35 +910,72 @@ makeSuite('AMPL aToken', (testEnv: TestEnv) => {
...
 
+      // console.log((await ampl.balanceOf(lenderAAddress)).toString());
+      // console.log((await aAMPL.scaledBalanceOf(lenderAAddress)).toString());
+      // console.log((await aAMPL.balanceOf(lenderAAddress)).toString());
+      // console.log((await aAMPL.totalSupply()).toString());
+      // console.log((await ampl.balanceOf(reserveData.aTokenAddress)).toString());
+      // console.log((await ampl.balanceOf(borrowerAAddress)).toString());
+      // console.log((await debtToken.balanceOf(borrowerAAddress)).toString());
+      // console.log((await aAMPL.balanceOf(treasuryAddress)).toString());
+      // console.log("------")
```
Comment (ahnaguib) [inline test/ampl-atoken.spec.ts]: remove the commented logs?

File: test/ampl-atoken.spec.ts [hunk 3]
```diff
@@ -43,14 +43,14 @@ async function rebase(pool, ampl, perc){
 
   // Interest rate changes up or down after rebase
   // Option 1: tell lending pool to recalculate rate
-  await ampl.rebase(1, supplyDelta.toString(10));
+  // await ampl.rebase(1, supplyDelta.toString(10));
   // await pool.syncInterestRates(ampl.address);
```
Comment (ahnaguib) [inline test/ampl-atoken.spec.ts]: Remove the commented part?

**Review by ahnaguib — APPROVED** [review]

**Pull request merged**